src/repeat.js

```diff
@@ -94,17 +115,74 @@ export class Repeat {
     }
   }
 
-  processItemsByStrategy() {
+  getInnerCollection() {
+    let expression = unwrapExpression(this.sourceExpression);
+    if (!expression) {
+      return null;
+    }
+    return expression.evaluate(this.scope, null);
+  }
+
+  observeInnerCollection() {
+    let items = this.getInnerCollection();
+    if (items instanceof Array) {
+      this.collectionObserver = this.observerLocator.getArrayObserver(items);
+    } else if (items instanceof Map) {
+      this.collectionObserver = this.observerLocator.getMapObserver(items);
+    } else {
+      return false;
+    }
+    this.callContext = 'handleInnerCollectionChanges';
+    this.collectionObserver.subscribe(this.callContext, this);
+    return true;
+  }
```

Comment:
Yes, I think that would be nice. I wonder if we could only have the `observeCollection` method and set the items conditionally based on what `getInnerCollection` returns? In order to handle all collection type specifics in the strategies. Perhaps something like this (untested): 

```javascript
observeCollection() {
    let innerCollection = this.getInnerCollection();
    if (innerCollection) {
      items = innerCollection; 
      this.callContext = 'handleInnerCollectionChanges';
    } else {
      items = this.items;
      this.callContext = 'handleCollectionChanges';
    }       
    
    this.collectionObserver = this.collectionStrategy.getCollectionObserver(items);
    if (this.collectionObserver) {
      this.collectionObserver.subscribe(this.callContext, this);
    }
  }
```